acmed/src/main_event_loop.rs

```diff
@@ -180,14 +180,15 @@ async fn renew_certificate(
 	endpoint_s: EndpointSync,
 ) -> (&mut Certificate, AccountSync, EndpointSync) {
 	loop {
-		match certificate.should_renew().await {
-			Ok(true) => break,
-			Ok(false) => {}
+		match certificate.schedule_renewal().await {
+			Ok(duration) => {
+				sleep(duration).await;
+				break;
+			}
 			Err(e) => {
 				certificate.warn(&e.message);
 			}
 		}
-		sleep(Duration::from_secs(crate::DEFAULT_SLEEP_TIME)).await;
```

Comment:
If `certificate.schedule_renewal()` returns an error, it is very likely to also return one in the next call, which will cause the loop to flood the logs with error. Therefore I think there is an interest in keeping the sleep.

Reply:
This is a good point, didn't think of that. I'd move the sleep into the error block though, so that it only sleeps more than required when something goes wrong.

Reply:
I've added a bit of exponential backoff here, does that seem good to you like it is now?

Reply:
Yes, all good, thank you! :smile: